### PR TITLE
Schema leak

### PR DESCRIPTION
**Pull request by joshmeranda**
We neglect to remove schemas from `s.schemas` in `s.doRemoveSchema` which results in duplicate entries in `s.schemas` which are never removed. This introduces a memory leak of ~300 mb a week on a large rancher cluster.

https://github.com/rancher/rancher/issues/47694

### CODE REVIEW

File: types/schemas.go [hunk 1]
```diff
@@ -84,6 +84,12 @@ func (s *Schemas) doRemoveSchema(schema Schema) *Schemas {
 		s.removeEmbed(&schema)
 	}
 
+	for i, candidate := range s.schemas {
+		if candidate.ID == schema.ID {
+			s.schemas = append(s.schemas[:i], s.schemas[i+1:]...)
```
Comment (aruiz14) [inline types/schemas.go:89]:
```
s.schemas = slices.DeleteFunc(s.schemas, func(candidate Schema) bool {
	return candidate.ID == schema.ID
})
```

**Review by aruiz14 — COMMENTED** [review]
Nice find!

File: types/schemas.go (hunk 1 above)
Comment (joshmeranda) [reply types/schemas.go]: Good call, thanks!

File: types/schemas_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSchemas(t *testing.T) {
```
Comment (aruiz14) [inline types/schemas_test.go:9]: Thanks for adding tests! However, this does not seem to cover the actual code that this PR is modifying, correct? Could you add a new case for that?

Comment (joshmeranda) [reply types/schemas_test.go:9]: It does, the code we added runs every time we run `AddSchema` on a schema which is an embedded type. In the test cases we embed the `hobbit` schema into the `baggins` schema which triggers the `doRemoveSchema` and removes the old `baggins` type while embedding.

**Review by aruiz14 — APPROVED** [review]

File: types/schemas_test.go (hunk 2 above)
Comment (aruiz14) [reply types/schemas_test.go:9]: Oh I missed it since it was further down in the call stack, thanks for the clarification!

Comment (joshmeranda) [reply types/schemas_test.go:9]: All good, thanks for being thorough!

**Review by nflynt — DISMISSED** [review]

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Since we're introducing `slices` now, the search code at lines 144-148 could be replaced with a call to `slices.IndexFunc`.

**Review by ericpromislow — APPROVED** [review]
Looks great

**Review by nflynt — APPROVED** [review]

**Review by crobby — APPROVED** [review]

**Pull request merged**